pkg/vault/contracts/RouterHooks.sol

```diff
@@ -0,0 +1,402 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IERC4626 } from "@openzeppelin/contracts/interfaces/IERC4626.sol";
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { IPermit2 } from "permit2/src/interfaces/IPermit2.sol";
+
+import { IWETH } from "@balancer-labs/v3-interfaces/contracts/solidity-utils/misc/IWETH.sol";
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+import "@balancer-labs/v3-interfaces/contracts/vault/RouterTypes.sol";
+
+import { RouterWethLib } from "./lib/RouterWethLib.sol";
+import { RouterCommon } from "./RouterCommon.sol";
+
+/**
+ * @notice Base Router with hooks for swaps and liquidity operations via Vault.
+ * @dev Implements hooks for init, add liquidity, remove liquidity, and swaps.
+ */
+contract RouterHooks is RouterCommon {
+    using Address for address payable;
+    using RouterWethLib for IWETH;
+    using SafeCast for *;
+
+    constructor(
+        IVault vault,
+        IWETH weth,
+        IPermit2 permit2,
+        string memory routerVersion
+    ) RouterCommon(vault, weth, permit2, routerVersion) {
+        // solhint-disable-previous-line no-empty-blocks
+    }
+
+    /**
+     * @notice Hook for initialization.
+     * @dev Can only be called by the Vault.
+     * @param params Initialization parameters (see IRouter for struct definition)
+     * @return bptAmountOut BPT amount minted in exchange for the input tokens
+     */
+    function initializeHook(
+        InitializeHookParams calldata params
+    ) external nonReentrant onlyVault returns (uint256 bptAmountOut) {
+        return _initializeHook(params);
```

Comment:
(here and elsewhere): is there a particular reason to have an internal version of the hooks if it's used only once, as opposed to just inline the code of the internal function here?

I'm not strongly against it, but if the internal function is only used once I'd say inlining it is better. Less boilerplate code around.

Reply:
That's fine then!